### PR TITLE
fixed: ignore singular point in mixed LR ASM evalSolution

### PR DESCRIPTION
**Pull request by akva2**

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this with downstreams please

**Comment by akva2** [comment]
jenkins build this with downstreams please

File: src/ASM/ASMs2DSpec.C [hunk 1]
```diff
@@ -307,6 +307,7 @@ bool ASMs2DSpec::evalSolution (Matrix& sField, const IntegrandBase& integrand,
 
 	// Compute the Jacobian inverse
 	fe.detJxW = utl::Jacobian(Jac,fe.dNdX,Xnod,dNdu);
+        if (fe.detJxW == 0.0) continue; // skip singular points
```
Comment (kmokstad) [inline src/ASM/ASMs2DSpec.C:310]: Wrong indent

File: src/ASM/ASMs2DLag.C [hunk 2]
```diff
@@ -781,6 +781,7 @@ bool ASMs2DLag::evalSolution (Matrix& sField, const IntegrandBase& integrand,
 
 	// Compute the Jacobian inverse
 	fe.detJxW = utl::Jacobian(Jac,fe.dNdX,Xnod,dNdu);
+    if (fe.detJxW == 0.0) continue; // skip singular points
```
Comment (kmokstad) [inline src/ASM/ASMs2DLag.C:784]: wrong indent

File: src/ASM/ASMs3DLag.C [hunk 3]
```diff
@@ -1074,6 +1074,7 @@ bool ASMs3DLag::evalSolution (Matrix& sField, const IntegrandBase& integrand,
 
 	  // Compute the Jacobian inverse
 	  fe.detJxW = utl::Jacobian(Jac,fe.dNdX,Xnod,dNdu);
+          if (fe.detJxW == 0.0) continue; // skip singular points
```
Comment (kmokstad) [inline src/ASM/ASMs3DLag.C:1077]: wrong indent

File: src/ASM/ASMs3DSpec.C [hunk 4]
```diff
@@ -455,6 +455,7 @@ bool ASMs3DSpec::evalSolution (Matrix& sField, const IntegrandBase& integrand,
 
 	  // Compute the Jacobian inverse
 	  fe.detJxW = utl::Jacobian(Jac,fe.dNdX,Xnod,dNdu);
+          if (fe.detJxW == 0.0) continue; // skip singular points
```
Comment (kmokstad) [inline src/ASM/ASMs3DSpec.C:458]: wrong indent

**Review by kmokstad — CHANGES_REQUESTED** [review]
Too quick copy-pasting, I persume..

File: src/ASM/ASMs2DSpec.C (hunk 1 above)
Comment (akva2) [reply src/ASM/ASMs2DSpec.C:310]: it's not, the indents are correct, the surrounding code uses tabs..

**Review by kmokstad — APPROVED** [review]

**Pull request merged**